Compilation fix

ge0rdi opened the pull request:
This will fix KPH compilation.
And also PH compilation for ARM64.

dmex commented on phnt/include/ntbcd.h:
```diff
@@ -21,6 +21,8 @@
 #ifndef _NTBCD_H
 #define _NTBCD_H
 
+#include <minwindef.h>
```
Why does this need to be imported?

ge0rdi replied:
Because of `CALLBACK` definition.

dmex replied:
I've deleted the callback type.

dmex commented on phnt/include/ntexapi.h:
```diff
@@ -22,6 +22,7 @@
 #define _NTEXAPI_H
 
 #include <ntkeapi.h>
+#include <ntimage.h>
```
Why does this also need to be imported?

ge0rdi replied:
Because of `PIMAGE_EXPORT_DIRECTORY` used in `SYSTEM_GDI_DRIVER_INFORMATION` structure.

ge0rdi replied:
Hmm, now I noticed that `ntimage.h` is only part of WDK.
And PHNT's `ntexapi.h` is used also in ProcessHacker (non-WDK) project.
So this change will make PH not compilable without WDK.

Will that be a problem?
Or WDK is required either way?

ge0rdi replied:
Eventually we can use forward declaration instead of `ntheader.h` include:
`typedef struct _IMAGE_EXPORT_DIRECTORY * PIMAGE_EXPORT_DIRECTORY;`

What do you think?

dmex replied:
We don't use that structure so I changed the type

dmex commented:
I've fixed the other issues. The only change required is the one for `!defined(_ARM64_))`

ge0rdi commented:
> I've fixed the other issues. The only change required is the one for !defined(_ARM64_))

Thank you.

Updated PR.
Now it just fixes ARM64 build.

dmex commented:
> Now it just fixes ARM64 build.

Merged.

Btw.. These !_ARM64_ checks did work but regressed after we enabled the standards conforming compiler so I guess defined() is the only reliable check.
